numpy/lib/arraysetops.py

```diff
@@ -240,58 +262,77 @@ def reshape_uniq(uniq):
         return uniq
 
     output = _unique1d(consolidated, return_index,
-                       return_inverse, return_counts)
-    if not (return_index or return_inverse or return_counts):
+                       return_inverse, return_counts,
+                       return_flags, return_data)
+    if not (return_index or return_inverse or return_counts
+            or return_flags or not return_data):
         return reshape_uniq(output)
-    else:
+    elif return_data:
         uniq = reshape_uniq(output[0])
         return (uniq,) + output[1:]
+    else:
+        return output
 
 def _unique1d(ar, return_index=False, return_inverse=False,
-              return_counts=False):
+              return_counts=False, return_flags=False,
+              return_data=True):
     """
     Find the unique elements of an array, ignoring shape.
     """
     ar = np.asanyarray(ar).flatten()
 
     optional_indices = return_index or return_inverse
-    optional_returns = optional_indices or return_counts
+    optional_returns = (optional_indices or return_counts
+                        or return_flags or not return_data)
 
     if ar.size == 0:
         if not optional_returns:
             ret = ar
         else:
-            ret = (ar,)
+            if return_data:
+                ret = (ar,)
```

Comment:
This would be cleaner as another `+=` like the others, and starting with `ret = ()` - even if it's (negligeably) slower